daemon/tests/harness/flow.rs

```diff
@@ -62,15 +37,56 @@ pub async fn next<T>(rx: &mut watch::Receiver<T>) -> Result<T>
 where
     T: Clone,
 {
-    rx.changed()
+    next_with(rx, |v| Some(v)).await
+}
+
+/// Returns watch channel value upon change
+pub async fn next_with<T, U>(
+    rx: &mut watch::Receiver<T>,
+    filter_map: impl Fn(T) -> Option<U>,
+) -> Result<U>
+where
+    T: Clone,
+{
+    let wait_until_predicate = async {
+        loop {
+            rx.changed().await?;
+
+            let current = rx.borrow().clone();
+
+            match filter_map(current) {
+                Some(val) => return anyhow::Ok(val),
+                None => {
+                    rx.changed().await?;
```

Comment:
❓ Why don't we continue onto the next iteration of the loop and wait for `changed` there?

Reply:
So it turns out that the semantics of waiting for the next change are still useful for some tests where we don't use the macro.

Reply:
Ah you are right. That doesn't make any sense!